eos-tech-support/eos-write-installer

```diff
@@ -316,8 +316,8 @@ if [ -f "$INSTALLER_IMAGE" ]; then
     # At this point udev should have seen that sfdisk closed the block device
     # which was opened for writing, so it will be reprobing.
     # Add udevadm settle here to avoid races.
-    udevadm settle
```

Comment:
I think there is a reason to leave this settle in this position (in addition to adding the one 2 lines down). I think in response to sfdisk closing the block device, udev will momentarily remove and then recreate the block device. That would race with partprobe trying to open it on the following line. The race is avoided with this settle, making sure udev has finished doing its thing.

Reply:
Revised to do both, and update the comment(s).